browser/script_injector/script_injector_browsertest.cc

```diff
@@ -0,0 +1,163 @@
+#include "mojo/public/cpp/bindings/associated_remote.h"
+#include "net/dns/mock_host_resolver.h"
+#include "net/test/embedded_test_server/embedded_test_server.h"
+#include "third_party/blink/public/common/associated_interfaces/associated_interface_provider.h"
+
+#if BUILDFLAG(IS_ANDROID)
+#include "chrome/test/base/android/android_browser_test.h"
+#else
+#include "chrome/test/base/in_process_browser_test.h"
+#endif
+
+namespace {
+
+const char kScript[] = R"(
+      (() => {
+        return new Promise((resolve) => {
+          document.title = 'test';
+          resolve(%s)
+        });
+      })();
+      )";
+
+}  // namespace
+
+mojo::AssociatedRemote<script_injector::mojom::ScriptInjector> GetRemote(
+    content::RenderFrameHost* rfh) {
+  mojo::AssociatedRemote<script_injector::mojom::ScriptInjector>
+      script_injector_remote;
+  rfh->GetRemoteAssociatedInterfaces()->GetInterface(&script_injector_remote);
+  return script_injector_remote;
+}
+
+std::unique_ptr<net::test_server::HttpResponse> HandleRequest(
+    const net::test_server::HttpRequest& request) {
+  auto response = std::make_unique<net::test_server::BasicHttpResponse>();
+  response->set_code(net::HTTP_OK);
+  response->set_content(R"(
+                          <html>
+                           <head><title>OK</title></head>
+                          </html>
+                        )");
+  response->set_content_type("text/html; charset=utf-8");
+  return response;
+}
+
+class ScriptInjectorBrowserTest : public PlatformBrowserTest {
+ public:
+  ScriptInjectorBrowserTest()
+      : https_server_(net::EmbeddedTestServer::TYPE_HTTPS) {}
+
+  void SetUpOnMainThread() override {
+    PlatformBrowserTest::SetUpOnMainThread();
+
+    mock_cert_verifier_.mock_cert_verifier()->set_default_result(net::OK);
+    host_resolver()->AddRule("*", "127.0.0.1");
+    https_server_.RegisterRequestHandler(base::BindRepeating(HandleRequest));
+    content::SetBrowserClientForTesting(&client_);
+
+    ASSERT_TRUE(https_server_.Start());
+  }
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    PlatformBrowserTest::SetUpCommandLine(command_line);
+    mock_cert_verifier_.SetUpCommandLine(command_line);
+  }
+
+  void SetUpInProcessBrowserTestFixture() override {
+    PlatformBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
+    PlatformBrowserTest::TearDownInProcessBrowserTestFixture();
+  }
+
+  content::WebContents* web_contents() {
+    return chrome_test_utils::GetActiveWebContents(this);
+  }
+
+ protected:
+  net::EmbeddedTestServer https_server_;
+
+ private:
+  BraveContentBrowserClient client_;
+  content::ContentMockCertVerifier mock_cert_verifier_;
+};
+
+// TESTS
+
+IN_PROC_BROWSER_TEST_F(ScriptInjectorBrowserTest, ScriptInjected) {
```

Comment:
ScriptInjector can be used without awaiting a promise so maybe this should be `InjectScriptAwaitPromise` or something and then similar for the others?